### PR TITLE
optional Duration formatting

### PR DESCRIPTION
**Pull request by Shoelace**
okay a few things in this pull..
you may only want to take bits and peices of it.

new option (durationformat) to control how durations are show.. mainly on progress bars
estimateupgradesecs() now returns a duration object which is then filtered by 'duration' filter
3 options 
* human - oprginal humanise call
* full - a full duraing from years to seconds
* abbreviated - only show say 2 or 3 levels of details.. ie years and months.. or days and hours

abbreviated does not work just yet.

i also have $scope.duration_examples  in eth controller as i couldnt get literal durations to format on the options screen for examples.

also a minor change to debug to change 50x for 60x and add a  3600x  so 1 sec -> 1 min or 1 hr

### CODE REVIEW

File: app/scripts/directives/unit.coffee [hunk 1]
```diff
@@ -22,11 +22,8 @@ angular.module('swarmApp').directive 'unit', ($log, game, commands, options, uti
       #util.utcdoy 1000 * secs
       if isFinite estimate.val
         nonlinear = if not (estimate.unit?.isVelocityConstant?() ? true) then 'less than ' else ''
```
Comment (erosson) [inline app/scripts/directives/unit.coffee]: I commented before about prefixing durations with 'less than' if the estimate's not exact (involves units-producing-units). That's still important; I don't want people reporting incorrect times as bugs. (Hopefully I'll improve the estimates soon so it's not necessary.)

File: app/views/directive-unit.html [hunk 2]
```diff
@@ -144,7 +144,7 @@
 
         <div class="progress" style="margin-bottom:0">
           <div class="progress-bar" role="progressbar" aria-valuenow="{{abil.costMetPercent().times(1000)}}" aria-valuemin="0" aria-valuemax="1000" style="width:{{abil.costMetPercent()|percent:1}}">
-            {{abil.costMetPercent()|percent:{floor:true} }}{{estimtateUpgradeSecs(abil)}}
+            {{abil.costMetPercent()|percent:{floor:true} }} {{estimtateUpgradeSecs(abil)|warpDuration}}
```
Comment (erosson) [inline app/views/directive-unit.html]: Why's this one different?

**Comment by erosson** [comment]
Haven't had a chance to run/merge this yet, but reading through it looks pretty good, except for broken tests and incorrect estimates involving meat-units. Hopefully I'll finish #304 soon so "less than" times aren't needed.

File: app/scripts/directives/unit.coffee (hunk 1 above)
Comment (Shoelace) [inline app/scripts/directives/unit.coffee]: Yes. I was wondering if I could not annotate the duration object with the nonlinear value such that the filter could access it. 

The other option would be to expose the nonlinear function and prefix it where required.

File: app/views/directive-unit.html (hunk 2 above)
Comment (Shoelace) [inline app/views/directive-unit.html]: I think that was just me testing both options

**Comment by Shoelace** [comment]
Thanks

I need to look into the test stuff. I haven't put that on any of my code. 

I'll try to learn that and start adding the tests. 

For 304 I assume we end up with a geometric progression so we would need the 2nd derivative of the velocity. Ie the acceleration of the acceleration    Which I think should be mostly constant for a given set of producers.

**Comment by Shoelace** [comment]
this should be all good now..  'less than' is working with an annotated duration object.
unit tests are not correct also.

**Comment by erosson** [comment]
Great, thanks! Ran out of time to test this tonight but I'll try to merge tomorrow.

**Comment by erosson** [comment]
going to remove the options UI for this before releasing. Here's why:
* if we get good estimates, the "human" format is pretty useless - everyone would like more precise estimates. numbers ticking downwards looks cooler than "a few seconds".
* I've worked on improving estimates enough to be sure that the math for large estimates, as the duration increases, will either get slower or less-precise - pick one. not sure how much slower yet - if it's slow enough that I have to pick "less-precise", I'd rather remove the "full" option for showing precise large durations now, than remove it later after people have started using it. also don't want to show estimates that keep changing/wrong estimates. should leave the code in for this though, just removing UI - if better estimates *are* fast enough, could bring this back.

removing these two leaves only one option, abbrev'd format. Does this seem reasonable? Sorry to throw away some of your work.

other small changes needed:
* seconds-only is shown as an unannotated number, "13", no indication that it's seconds. "0:13" would be better.
* infinite time left (zero production) is shown as "0".

will work on these soon if you don't beat me to it - done for tonight. (sorry this is taking so long)

this is good though, thank you! I hadn't realized passing `nonlinear` to the filter would be awkward; good solution

**Comment by Shoelace** [comment]
Leave it with me you seem busy with other changes. But I may not get to it until Monday. 

Remove the ui? Or just hide behind debug mode ?

I'll set abbreviated to default and set it so that it shows
Under 1 day: hh:mm:ss
More the 1 day: 2 levels of detail 
Ie. Year/month .  Month/day or day/ hours

I'll try and turn 0 production into "never" infinity or just blank

**Comment by erosson** [comment]
Merged, thanks for being patient with me on this one!

**Pull request merged**